feat: Update corpay fields logic, integrate API

pasyukevich opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Explanation of Change
<!-- Explain what your change does and how it addresses the linked issue -->

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/50904, https://github.com/Expensify/App/issues/50903
PROPOSAL:


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image

It's acceptable to write "Same as tests" if the QA team is able to run the tests in the above "Tests" section.
--->
// TODO: These must be filled out, or the issue title must include "[No QA]."


1. Change workspace currency to EUR, GBP, CAD or AUD
2. Enable workflows for the workspace
3. Open workflows
4. Try to connect a bank account
5. Go to the second step - a new component should be displayed
6. For  the first substep validations and field lists will be received from the BE
7. For the AUD currency we should have 3 substeps (on the second upload file step, for other currencies should be skipped)
8. Verify Upload picker validations (10 files limit, 5MB total, PDF/PNG/JPEG only)
9. Go to the third step 
10. Repeat steps and verify that error is showed due to invalid data 

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

[android-native-converted.webm](https://github.com/user-attachments/assets/520e1b94-5c80-4c47-bed7-3ecc76d0ad44)


</details>

<details>
<summary>Android: mWeb Chrome</summary>

[android-web-converted.webm](https://github.com/user-attachments/assets/4de8314e-91ca-4ea1-8d3a-29bd59f2805d)


</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/8209865b-510f-4684-b672-67be75f9ae77




</details>

<details>
<summary>iOS: mWeb Safari</summary>


https://github.com/user-attachments/assets/7474cbd6-dabc-4c05-8744-9e971c9af881



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


https://github.com/user-attachments/assets/39b08f48-d7f1-47be-a5c6-e08494be3150



</details>

<details>
<summary>MacOS: Desktop</summary>


https://github.com/user-attachments/assets/ca6e14c0-bb72-49ca-a3b0-9cb12a5f1ac4



</details>

pasyukevich committed:
feat: Update corpay fields logic, integrate API

pasyukevich committed:
Merge branch 'main' into feature/GR-step-two-api-integration

pasyukevich committed:
fix: variable value

pasyukevich committed:
feat: Add Corpay bank account creation parameters and update related components

pasyukevich committed:
fix: Update variable name for destination country in currency selection

pasyukevich committed:
feat: Enhance Corpay bank account creation with loading and success states

pasyukevich committed:
feat: Add loading and success states for Corpay bank account fields

pasyukevich committed:
Merge branch 'main' into feature/GR-step-two-api-integration

pasyukevich committed:
Merge branch 'main' into feature/GR-step-two-api-integration

pasyukevich committed:
Merge branch 'main' into feature/GR-step-two-api-integration

pasyukevich committed:
Refactor reimbursement account state management to replace isSuccess with isCreateCorpayBankAccount flag

pasyukevich committed:
Merge branch 'main' into feature/GR-step-two-api-integration

pasyukevich committed:
Update title handling for account holder country in confirmation step

pasyukevich committed:
Refactor reimbursement account handling to include isSuccess flag and update corpayFields structure

pasyukevich committed:
Merge branch 'main' into feature/GR-step-two-api-integration

pasyukevich committed:
Remove unused parameter 'preferredMethod' from Confirmation component

pasyukevich committed:
Remove redundant condition from reimbursement account success check in Confirmation component

pasyukevich committed:
Add isLoading flag cleaning

pasyukevich committed:
Update formID reference in Confirmation component and adjust styles

pasyukevich committed:
Add margin styles to submit button in Confirmation component

pasyukevich committed:
Merge branch 'main' into feature/GR-step-two-api-integration

hungvu193 commented:
Can you resolve conflicts? @pasyukevich

hungvu193 commented:
I think we should clear the error if we closed the RHP

https://github.com/user-attachments/assets/8c518090-067a-4758-b538-d36e8ad00bb8

hungvu193 commented on src/pages/ReimbursementAccount/NonUSD/Country/substeps/Confirmation.tsx:
```diff
@@ -80,13 +80,16 @@ function Confirmation({onNext}: SubStepProps) {
                     />
                     <Text style={[styles.ph5, styles.mb3, styles.mutedTextLabel]}>
-                        {`${translate('countryStep.yourBusiness')} ${translate('countryStep.youCanChange')}`}{' '}
-                        <TextLink
-                            style={[styles.label]}
```
If we don't use TextLink, it will again cause the misaligned text on IOS

![simulator_screenshot_09004549-4156-47BA-A52F-A491576132C1](https://github.com/user-attachments/assets/4905f416-4078-4c82-9e06-cef52ccedd22)

pasyukevich replied:
Updated

pasyukevich committed:
Merge branch 'main' into feature/GR-step-two-api-integration

pasyukevich committed:
Add loading indicator to BankAccountDetails and update Confirmation step with TextLink

hungvu193 commented:
@pasyukevich Let me know when it's ready for another review 😄

pasyukevich committed:
Merge branch 'main' into feature/GR-step-two-api-integration

pasyukevich commented:
@hungvu193 sure 

Currently working on app crash on the reimbursement flow after yesterday's merge

![image](https://github.com/user-attachments/assets/07a95f33-32b4-4046-8eb5-03f8f940149f)

pasyukevich committed:
Merge branch 'main' into feature/GR-step-two-api-integration

pasyukevich committed:
fix: eslint,  policyID update to possible undefined

pasyukevich committed:
fix:  COUNTRY field to reimbursement account form and update related components

pasyukevich commented:
The problem with the crash went after a couple of merges, I've addressed eslint errors after them, 

Fixed issue with county ISO
I will also update the flow to clean the form after currency change since it has become crucial for other steps

pasyukevich committed:
fix: rollback redundant changes

pasyukevich committed:
fix: clear draft values for reimbursement account form on currency selection

pasyukevich committed:
Merge branch 'main' into feature/GR-step-two-api-integration

pasyukevich commented:
@hungvu193 PR is ready for rereview

pasyukevich commented:
Regarding the Onyx error - it is coming from the BE 

We are receiving an empty array while expecting an object

![image](https://github.com/user-attachments/assets/c4c08712-49e2-4131-b73b-2de1438ae53f)

<img width="581" alt="image" src="https://github.com/user-attachments/assets/d5fb9c72-9a15-4640-afe9-739bf6ab04c8" />

<img width="581" alt="image" src="https://github.com/user-attachments/assets/52303858-7b70-4ba3-a9eb-e8e14fbb9d95" />

hungvu193 commented:
Can you add the gap below `Beneficiary Account Number `?

<img width="362" alt="Screenshot 2024-12-21 at 23 46 12" src="https://github.com/user-attachments/assets/d6ac3c91-139d-4498-9a3f-ce84b41c4374" />

hungvu193 commented:
Also, I think I can still reproduce this one: https://github.com/Expensify/App/pull/52812#issuecomment-2545310360

pasyukevich committed:
fix: clear form errors on confirmation step and update country field description

pasyukevich commented:
> Also, I think I can still reproduce this one: [#52812 (comment)](https://github.com/Expensify/App/pull/52812#issuecomment-2545310360)

Fixed

pasyukevich committed:
fix: remove unnecessary flex style from bank account details components

hungvu193 commented:
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

https://github.com/user-attachments/assets/8a778373-726d-4595-8ec7-7ea3088c3a48

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

https://github.com/user-attachments/assets/2e624ddf-6579-4fa3-9583-935d10264a6e

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

https://github.com/user-attachments/assets/4504b0f3-838f-4217-9841-28a5a9494d47

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->


https://github.com/user-attachments/assets/4798567b-63f4-4b8b-8bf7-9b764eab95b7

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

https://github.com/user-attachments/assets/0cd11b47-3e23-43e8-a7e1-00619254204a

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

https://github.com/user-attachments/assets/f8492c83-aa1b-47ae-af87-ef243517cd55

</details>

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

pasyukevich commented on src/libs/API/parameters/VerifyIdentityForBankAccountParams.ts:
```diff
@@ -1,6 +1,6 @@
 type VerifyIdentityForBankAccountParams = {
     bankAccountID: number;
     onfidoData: string;
-    policyID: string;
+    policyID?: string;
 };
```
How should we handle this case then? 
https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#case-1-argument-of-type-string--undefined-is-not-assignable-to-parameter-of-type-string

We have the example exactly with `policyID`
